### ISSUE TITLE
Change in indexing behaviour with numpy ints

### ISSUE THREAD

**Issue by benjeffery** [zarr-developers/zarr-python#967]
After https://github.com/zarr-developers/zarr-python/commit/7c31f041ca8baab860564d9c6332c52f19777e80 there was a breaking change to indexing behavior when a numpy integer type is passed:

For the following array:
```python
z = zarr.group()
a = z.create_dataset("test",shape=(10,))
a[:]=range(10)
```
Before this commit:
`a[np.int64(1)]` has the value `1.0`
After:
`a[np.int64(1)]` has the value `[1.]`

For both cases `a[1]` stays as `1.0` as expected. Was this change expected? This happens a lot when iterating over one numpy array and indexing with it as the iterable is of the numpy type. The fix in our application was to cast to a python `int` before indexing.

EDIT: Also note that numpy indexing returns the single value as was the case before this commit.

**Comment by raphaelquast** [comment]
I was just about to write the exact same issue...   (stumbled upon it while trying to read an encoded dict)

```python
root = zarr.group()
root.create("asdf", shape=(10,), dtype=object, object_codec=numcodecs.JSON())
root.asdf[0] = dict(a=1, b=2)

# using a normal python int to get the data
root.asdf[0]
#>>> {'a': 1, 'b': 2}

# using a numpy int to get the data
ids = np.array([0, 1, 2])
root.asdf[ids[0]]
#>>> array([{'a': 1, 'b': 2}], dtype=object)
```

**Comment by benjeffery** [comment]
> This issue has been fixed in patch release [2.11.1](https://zarr.readthedocs.io/en/stable/release.html#release-2-11-1). tada

Woop! Thankyou for the quick release!
